equinox/nn/_embedding.py

```diff
@@ -1,17 +1,17 @@
-from typing import Optional
+from typing import Any, Optional
 
 import jax
 import jax.numpy as jnp
 import jax.random as jrandom
-from jaxtyping import Array, ArrayLike, Complex, Float, Int, PRNGKeyArray
+from jaxtyping import Array, ArrayLike, Float, Int, PRNGKeyArray
```

Comment:
Reason to remove usage of Complex: if we are trying to preserve/use certain dtypes, complex is just not expressive enough: e.g. there are no complex-bf16, so it is impossible to use it without a forceful type conversion.